src/app/models/ability.rb

```diff
@@ -0,0 +1,35 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    
+    user ||= User.new # nil guard; creats a new user object
+    if user.role?('admin')
+      can :manage, :all
+    elsif user.role?('teacher')
+      can :index, :dashboard
```

Comment:
jap we need to add further permissions later.

can :index, :dashboard means that the role is allowed to invoke the index method of the DashboardController which in this case provides the dashboard view